api/http-api.md

```diff
@@ -0,0 +1,173 @@
+# HTTP API
+
+## Overview
+
+Temporal has a gRPC API but no HTTP API. We need an HTTP API.
+
+**Goals**
+
+* Create HTTP API that governs all reasonable user-facing calls
```

Comment:
> Can we make sure this HTTP API work does not make it harder to support grpc streaming in future

It can make it harder depending on the shape of the stream. https://github.com/grpc-ecosystem/grpc-gateway now handles streams as newline-delimited JSON streams of request/response body but there are surely caveats.

> Most likely the polling API and the get history API could be candidate for streaming.

While they _could_, we wouldn't get rid of existing API surely because there are many relying on it across many SDK versions. So does streaming really need to stop the HTTP API project?

There is also a lot of detail that goes into proper failure handling and LB/HA of streams anyways that has to be discussed when we get there. I don't think this project needs to concern itself w/ future streaming possibility (nor can it really do anything about it short of contrib'ing something if we think it's messing). Maybe we want to make a rule "no streaming API without a non-stream equivalent".

We're using a third party library. I don't think there's anything to do here concerning some hypothetical feature if/until that feature has a more concrete spec.